packages/react/src/ReactElement.js

```diff
@@ -88,6 +88,25 @@ function defineRefPropWarningGetter(props, displayName) {
   });
 }
 
+function setProps(config, props) {
+  for (let propName in config) {
+    if (
+      hasOwnProperty.call(config, propName) &&
+      !RESERVED_PROPS.hasOwnProperty(propName)
+    ) {
+      props[propName] = config[propName];
+    }
+  }
+}
+
+function setDefaultProps(defaultProps, props) {
+  for (let propName in defaultProps) {
+    if (props[propName] === undefined) {
```

Comment:
question/nit: could this just be
```suggestion
    if (props[propName] == undefined) {
```

Reply:
I think here we just care if the prop value is `undefined`, otherwise explicitly setting a prop to `null` would fallback to the default prop since `null == undefined` is `true`. (perhaps it might be more clear to do `typeof props[propName] === "undefined"`)